lib/Model/TypeDefinition.cpp

```diff
@@ -115,17 +115,42 @@ model::Identifier model::StructField::name() const {
   return Result;
 }
 
-model::Identifier model::Argument::name() const {
-  Identifier Result;
+std::optional<model::Identifier>
+model::CABIFunctionDefinition::tryGetArgumentName(const Argument &Argument,
+                                                  const TypeOfGlobalNameCache
+                                                    &Names,
+                                                  const NamingConfiguration
+                                                    &Configuration) const {
+  model::Identifier Result;
+
+  if (Argument.CustomName().empty()) {
+    const std::string &Prefix = Configuration.unnamedFunctionArgumentPrefix();
+    (Prefix + llvm::Twine(Argument.Index())).toVector(Result);
+
+    for (const auto &AnotherArgument : Arguments())
+      if (not AnotherArgument.CustomName().empty())
+        if (AnotherArgument.CustomName() == Result)
+          return std::nullopt; // Argument name collides with another argument!
```

Comment:
This is not valid in the model.
My expectations is that we won't need any `tryGet...Name`, they are supposed to *always* succeed (on a valid model, which we can assume).